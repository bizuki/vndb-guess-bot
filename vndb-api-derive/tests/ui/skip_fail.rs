use vndb_api_derive::VndbFieldsEnum;

#[derive(VndbFieldsEnum)]
pub struct Struct {
    nested: Nested,
    #[vndb_field(skip)]
    skipped: String
}

#[derive(VndbFieldsEnum)]
pub struct Nested {
    field_a: String,
    field_b: String,
    #[vndb_field(skip)]
    nested: NestedNested
}

#[derive(VndbFieldsEnum)]
pub struct NestedNested {
    test: String
}

fn main() {
    let _ = StructFields::Skipped;
    let _ = StructFields::Nested(vec![NestedFields::FieldA]);
    let _ = StructFields::Nested(vec![NestedFields::FieldB]);
    let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test])]);
    let _ = StructFields::Nested(vec![NestedFields::FieldA, NestedFields::FieldB]);
    let _ = StructFields::Nested(vec![NestedFields::FieldA, NestedFields::Nested(vec![NestedNestedFields::Test])]);
    let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test]), NestedFields::FieldB]);
    let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test]), NestedFields::FieldB]);
}
