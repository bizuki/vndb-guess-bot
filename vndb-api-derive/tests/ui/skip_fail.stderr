error[E0599]: no variant or associated item named `Skipped` found for enum `StructFields` in the current scope
  --> tests/ui/skip_fail.rs:24:27
   |
3  | #[derive(VndbFieldsEnum)]
   |          -------------- variant or associated item `Skipped` not found for this enum
...
24 |     let _ = StructFields::Skipped;
   |                           ^^^^^^^ variant or associated item not found in `StructFields`

error[E0599]: no variant or associated item named `Nested` found for enum `NestedFields` in the current scope
  --> tests/ui/skip_fail.rs:27:53
   |
10 | #[derive(VndbFieldsEnum)]
   |          -------------- variant or associated item `Nested` not found for this enum
...
27 |     let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test])]);
   |                                                     ^^^^^^ variant or associated item not found in `NestedFields`

error[E0599]: no variant or associated item named `Nested` found for enum `NestedFields` in the current scope
  --> tests/ui/skip_fail.rs:29:75
   |
10 | #[derive(VndbFieldsEnum)]
   |          -------------- variant or associated item `Nested` not found for this enum
...
29 |     let _ = StructFields::Nested(vec![NestedFields::FieldA, NestedFields::Nested(vec![NestedNestedFields::Test])]);
   |                                                                           ^^^^^^ variant or associated item not found in `NestedFields`

error[E0599]: no variant or associated item named `Nested` found for enum `NestedFields` in the current scope
  --> tests/ui/skip_fail.rs:30:53
   |
10 | #[derive(VndbFieldsEnum)]
   |          -------------- variant or associated item `Nested` not found for this enum
...
30 |     let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test]), NestedFields::FieldB]);
   |                                                     ^^^^^^ variant or associated item not found in `NestedFields`

error[E0599]: no variant or associated item named `Nested` found for enum `NestedFields` in the current scope
  --> tests/ui/skip_fail.rs:31:53
   |
10 | #[derive(VndbFieldsEnum)]
   |          -------------- variant or associated item `Nested` not found for this enum
...
31 |     let _ = StructFields::Nested(vec![NestedFields::Nested(vec![NestedNestedFields::Test]), NestedFields::FieldB]);
   |                                                     ^^^^^^ variant or associated item not found in `NestedFields`
